apollo-federation/src/sources/connect/variable.rs

```diff
@@ -190,6 +190,10 @@ pub(crate) enum VariableParseError<I> {
         location: Range<usize>,
     },
     Nom(I, ErrorKind),
+    InvalidHeaderValue {
```

Comment:
This doesn't belong here - variables are agnostic to where they are used, so adding a header-specific variant here doesn't make sense. I think you added this because you need to propagate this error our of the HeaderValue parser and its using the same error type. That was just laziness on my part - at the time, the HeaderValue parser didn't *have* any errors that weren't variable errors, so I just passed back the variable error type. But since you're now adding an error that is specfiic to headers, you should add a new header error type and add this variant there instead.